scripts/analytics/benchmarks.py

```diff
@@ -138,7 +139,138 @@ def get_log_counts(users):
     return rows
 
 
+def get_projects():
+    projects = Node.find(
+        Q('category', 'eq', 'project') &
+        Q('is_deleted', 'eq', False) &
+        Q('is_folder', 'ne', True)
+    )
+    return projects
+
+def get_projects_forked():
+    projects_forked = list(Node.find(
+        Q('category', 'eq', 'project') &
+        Q('is_deleted', 'eq', False) &
+        Q('is_folder', 'ne', True) &
+        Q('is_fork', 'eq', True)
+    ))
+    return projects_forked
+
+def get_projects_registered():
+    projects_registered = Node.find(
+        Q('category', 'eq', 'project') &
+        Q('is_deleted', 'eq', False) &
+        Q('is_folder', 'ne', True) &
+        Q('is_registration', 'eq', True)
+    )
+    return projects_registered
+
+def get_projects_public():
+    projects_public = Node.find(
+        Q('category', 'eq', 'project') &
+        Q('is_deleted', 'eq', False) &
+        Q('is_folder', 'ne', True) &
+        Q('is_public', 'eq', True)
+    )
+    return projects_public
+
+def get_number_downloads_total(projects=None):
+    number_downloads_total = 0
+
+    if projects is None:
+        projects = get_projects()
+
+    contributors_per_project = []
+
+    contrib = {}
+
+    for project in projects:
+        contributors_per_project.append(len(project.contributors))
+        for person in project.contributors:
+            if not person:
+                continue
+            if person._id not in contrib:
+                contrib[person._id] = []
+            for neighbor in project.contributors:
+                if not neighbor:
+                    continue
+                if neighbor._id not in contrib[person._id]:
+                    contrib[person._id].append(neighbor._id)
+
+        addon = project.get_addon('osfstorage')
+
+        for filenode in OsfStorageFileNode.find(Q('node_settings', 'eq', addon) & Q('kind', 'eq', 'file')):
+            for idx, version in enumerate(filenode.versions):
+                page = ':'.join(['download', project._id, filenode._id, str(idx)])
+                unique, total = get_basic_counters(page)
+                number_downloads_total += total or 0
+
+        for filenode in OsfStorageTrashedFileNode.find(Q('node_settings', 'eq', addon) & Q('kind', 'eq', 'file')):
+            for idx, version in enumerate(filenode.versions):
+                page = ':'.join(['download', project._id, filenode._id, str(idx)])
+                unique, total = get_basic_counters(page)
+                number_downloads_total += total or 0
+
+    return number_downloads_total
+
+def get_number_downloads_unique(projects=None):
```

Comment:
This function has a tons of duplicated code as get_number_downloads_total. They can be simplified into one function.